src/main/java/domain/game/Judge.java

```diff
@@ -0,0 +1,72 @@
+package domain.game;
+
+import domain.player.Player;
+import domain.player.Players;
+import domain.prize.Prize;
+import domain.prize.Prizes;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Judge {
+    private static final String SEARCH_ALL_KEY_WORD = "all";
+    private final Players players;
+    private final Prizes prizes;
+    private final PathMapper pathMapper;
+    private final JudgeCache CACHE;
+
+    public Judge(final Players players, final Prizes prizes, final PathMapper pathMapper) {
+        this.players = players;
+        this.prizes = prizes;
+        this.pathMapper = pathMapper;
+        this.CACHE = new JudgeCache();
+    }
+
+    public Map<Player, Prize> search(final String name) {
+        if (SEARCH_ALL_KEY_WORD.equals(name)) {
+            return makeResultAll();
+        }
+
+        final Player target = getTarget(name);
+        return makeResult(target);
+    }
+
+    private Map<Player, Prize> makeResultAll() {
+        return this.players.getMembers()
+                .stream()
+                .collect(
+                        Collectors.toMap(
+                                player -> player,
+                                this::getPrizeMappedByLadder,
+                                (a, b) -> a,
+                                LinkedHashMap::new
+                        )
+                );
+    }
+
+    private Player getTarget(final String name) {
+        return this.players.findByName(name).orElseThrow(
+                () -> new IllegalArgumentException("사용자를 찾을 수 없습니다."));
+    }
```

Comment:
```suggestion
    private Player getTarget(final String name) {
        return this.players.findByName(name)
                .orElseThrow(() -> new IllegalArgumentException("사용자를 찾을 수 없습니다."));
    }
```

이런 개행은 어떨까요? 메소드 분리, 클래스 분리도 중요하지만 가장 먼저 역할이 잘 분리되어야 하는건 라인분리인 것 같아요!
하나의 라인에서 여러 역할을 하기보단 예외 처리를 위한 메소드 호출은 다음 라인에 역할을 주면 어떨까요?

Reply:
아하..! 플레이어를 찾는 역할과 예외 처리라는 역할, 2가지 역할이 있었군요. 꼼꼼하게 봐주셔서 감사해요 😀